lib/light-service/organizer/iterate.rb

```diff
@@ -1,21 +1,39 @@
+require 'light-service/action.rb'
+
 module LightService
   module Organizer
-    class Iterate
-      extend ScopedReducable
-
-      def self.run(organizer, collection_key, steps)
-        lambda do |ctx|
-          return ctx if ctx.stop_processing?
+    Iterate = Struct.new(:organizer, :collection_key, :steps) do
+      include ScopedReducable
 
-          collection = ctx[collection_key]
-          item_key = collection_key.to_s.singularize.to_sym
-          collection.each do |item|
-            ctx[item_key] = item
-            ctx = scoped_reduce(organizer, ctx, steps)
+      def rollback(ctx)
+        collection = ctx[collection_key]
+        item_key = collection_key.to_s.singularize.to_sym
+        collection.reverse_each do |item|
+          ctx[item_key] = item
+          steps.reverse_each do |action|
```

Comment:
```suggestion
          [steps].flatten.reverse_each do |action|
```

my 2 cent

from readme iterate allow `iterate(:retrieved_items, [TransformsData])`

from my test is possible to use

`iterate(:retrieved_items, TransformsData)`
`iterate(:retrieved_items, [TransformsData])`
`iterate(:retrieved_items, [TransformsData,TransformsData2])`

so the Array refinement can be excluded